src/qobj/quantum_object.jl

```diff
@@ -226,6 +232,11 @@ _check_dims(dims::Any) = throw(
     ),
 )
 
+function check_dims(A::AbstractQuantumObject, B::AbstractQuantumObject)
+    A.dims != B.dims && throw(DimensionMismatch("The two quantum objects don't have the same Hilbert dimension."))
+    return nothing
+end
```

Comment:
how about moving this function into `utilities.jl`, since it would be used by both `quantum_object.jl` and `quantum_object_evo.jl`